lib/bolt/transport/local.rb

```diff
@@ -83,14 +83,24 @@ def run_task(target, task, arguments, _options = {})
         implementation = task.select_implementation(target, PROVIDED_FEATURES)
         executable = implementation['path']
         input_method = implementation['input_method'] || 'both'
-
-        # unpack any Sensitive data, write it to a separate variable because
-        # we log 'arguments' below
-        unwrapped_arguments = unwrap_sensitive_args(arguments)
-        stdin = STDIN_METHODS.include?(input_method) ? JSON.dump(unwrapped_arguments) : nil
-        env = ENVIRONMENT_METHODS.include?(input_method) ? envify_params(unwrapped_arguments) : nil
+        extra_files = implementation['files']
 
         with_tmpscript(executable, target.options['tmpdir']) do |script, dir|
+          if extra_files
+            FileUtils.mkdir_p(extra_files.map { |file| File.join(dir, '_installdir', File.dirname(file['name'])) })
+            arguments['_installdir'] = File.join(dir, '_installdir')
```

Comment:
I don't think we're copying the task itself into the correct location. it looks like it will be outside the installdir only

Reply:
Oh, is the task supposed to be in installdir? I don't think we defined that.

Reply:
Pretty sure pxp-agent doesn't put it in installdir either.

Reply:
that was the original design but I'm not sure it's clear from the spec.